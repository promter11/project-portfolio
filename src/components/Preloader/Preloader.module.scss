@import "../../styles/_variables";
@import "../../styles/_mixins";

.preloader {
  position: fixed;
  top: 0;
  left: 0;
  z-index: 10;
  @include width-height(100vw, 100vh);
  transition: opacity 1s ease;
  background: $black-color;

  &.enter,
  &.exitActive {
    opacity: 1;
  }

  &.exit,
  &.enterActive {
    opacity: 0;
  }

  .block {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    @include flex(nowrap, row, center, center);
    animation: blur-text 2s infinite linear alternate;

    .letter {
      display: inline-block;
      margin: 0 0.5rem;
      color: $white-color;
    }
  }
}

@keyframes blur-text {
  0% {
    filter: blur(0);
  }

  100% {
    filter: blur(0.4rem);
  }
}
