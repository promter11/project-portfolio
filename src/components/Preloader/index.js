import React from "react";
import { bool } from "prop-types";
import { CSSTransition } from "react-transition-group";

import styles from "./Preloader.module.scss";

const Preloader = ({ loading }) => {
  return (
    <CSSTransition
      in={loading}
      timeout={1000}
      classNames={{
        enter: styles.enter,
        enterActive: styles.enterActive,
        exit: styles.exit,
        exitActive: styles.exitActive,
      }}
      unmountOnExit
    >
      <div className={styles.preloader}>
        <div className={styles.block}>
          {["L", "O", "A", "D", "I", "N", "G"].map((letter, index) => {
            return (
              <span key={index} className={styles.letter}>
                {letter}
              </span>
            );
          })}
        </div>
      </div>
    </CSSTransition>
  );
};

Preloader.propTypes = {
  loading: bool.isRequired,
};

export default Preloader;
