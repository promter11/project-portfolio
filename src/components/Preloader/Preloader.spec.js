import React from "react";
import { shallow } from "enzyme";

import Preloader from "./index";

const props = {
  loading: true,
};

describe("Preloader component", () => {
  it("renders correctly with props", () => {
    const wrapper = shallow(<Preloader {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
