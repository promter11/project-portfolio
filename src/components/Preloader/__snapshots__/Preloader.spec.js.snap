// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader component renders correctly with props 1`] = `
<CSSTransition
  classNames={
    Object {
      "enter": "enter",
      "enterActive": "enterActive",
      "exit": "exit",
      "exitActive": "exitActive",
    }
  }
  in={true}
  timeout={1000}
  unmountOnExit={true}
>
  <div
    className="preloader"
  >
    <div
      className="block"
    >
      <span
        className="letter"
        key="0"
      >
        L
      </span>
      <span
        className="letter"
        key="1"
      >
        O
      </span>
      <span
        className="letter"
        key="2"
      >
        A
      </span>
      <span
        className="letter"
        key="3"
      >
        D
      </span>
      <span
        className="letter"
        key="4"
      >
        I
      </span>
      <span
        className="letter"
        key="5"
      >
        N
      </span>
      <span
        className="letter"
        key="6"
      >
        G
      </span>
    </div>
  </div>
</CSSTransition>
`;
